src/components/DataTable/DataTableDataCell.tsx

```diff
@@ -19,10 +19,39 @@ const DataTableDataCell = ({
 }: DataCellProps) => {
   const anchorEl = React.useRef(null);
   const [showMenu, setShowMenu] = useState(false);
+  const [textContent, setTextContent] = useState("");
+
+  useEffect(() => {
+    let extractedText = "";
+    React.Children.forEach(children, (child) => {
+      if (typeof child === "string") {
+        extractedText += child;
+      } else if (React.isValidElement(child)) {
+        const childProps = child.props as {children?: React.ReactNode};
+        const childText = extractTextFromComponent(childProps.children);
+        extractedText += childText;
+      }
+    });
+    setTextContent(extractedText.trim());
+  }, [children]);
+
+  // Helper function to extract text from components
+  function extractTextFromComponent(component: React.ReactNode): string {
+    if (typeof component === "string") {
+      return component;
+    } else if (Array.isArray(component)) {
+      return component.map(extractTextFromComponent).join("");
+    } else if (React.isValidElement(component)) {
+      const childProps = component.props as {children?: React.ReactNode};
+      return extractTextFromComponent(childProps.children);
+    }
+    return "";
+  }
 
   return (
     <>
       <GcxDataTableDataCellRoot
+        title={textContent}
```

Comment:
Suggestion; it might be a simpler option for the cell to pass down a `title` prop ?

Reply:
as discussed, getting from cell value might include ids and information that we do not necessarily want to reveal